xsuite/src/world/csworld.test.ts

```diff
@@ -0,0 +1,801 @@
+import { beforeEach, expect, test, beforeAll, afterAll, assert } from "vitest";
+import { assertAccount, assertVs } from "../assert";
+import { e } from "../data";
+import { Tx } from "../proxy";
+import { DummySigner } from "./signer";
+import {
+  generateContractU8AAddress,
+  generateWalletU8AAddress,
+  isContractAddress,
+} from "./utils";
+import { CSWorld, CSContract, CSWallet } from ".";
+
+let world: CSWorld;
+let wallet: CSWallet;
+let otherWallet: CSWallet;
+let contract: CSContract;
+const fftId = "FFT-abcdef";
+const sftId = "SFT-abcdef";
+const worldCode = "file:contracts/world/output/world.wasm";
+const zeroBechAddress =
+  "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu";
+const zeroHexAddress =
+  "0000000000000000000000000000000000000000000000000000000000000000";
+const zeroBytesAddress = new Uint8Array(32);
+const emptyAccount = {
+  nonce: 0,
+  balance: 0,
+  code: "",
+  codeHash: "",
+  codeMetadata: "",
+  owner: "",
+  kvs: {},
+};
+const explorerUrl = "http://explorer.local";
+
+beforeAll(async () => {
+  world = await CSWorld.start({
+    explorerUrl,
+    // verbose: true,
+    // debug: true,
+    waitFor: 120_000,
+  });
+  wallet = await world.createWallet({
+    balance: 10n ** 18n,
+    kvs: [e.kvs.Esdts([{ id: fftId, amount: 10n ** 18n }])],
+  }); // wallet in shard 0
+  otherWallet = await world.createWallet();
+
+  // pass an epoch so system smart contracts are enabled
+  await world.generateBlocksUntilEpochReached(1);
+
+  contract = await wallet.createContract({
+    balance: 10n ** 18n,
+    code: worldCode,
+    codeMetadata: ["payable"],
```

Comment:
Pushed an improvement today in sworld.test.ts and in that test it is now `["readable"]` which is used. Will be good to have the same thing here.

Reply:
Should be done